src/raw_iter.rs

```diff
@@ -260,46 +318,39 @@ impl<'a> Iterator for Rle8Pixels<'a> {
 ///
 /// Each pixel is returned as a `u32` regardless of the bit depth of the source image.
 #[derive(Debug)]
-pub struct Rle4Pixels<'a> {
+pub struct Rle4Colors<'a> {
     /// Our source data
     data: &'a [u8],
     /// Our state
     rle_state: RleState,
-    /// The width of a line in pixels
-    width: u32,
-    /// The location of the next pixel
-    next_pixel: Point,
+    start_of_line: bool,
 }
 
-impl<'a> Rle4Pixels<'a> {
+impl<'a> Rle4Colors<'a> {
     /// Create a new RLE pixel iterator.
-    pub(crate) fn new(raw_bmp: &RawBmp<'a>) -> Rle4Pixels<'a> {
+    pub(crate) fn new(raw_bmp: &RawBmp<'a>) -> (Rle4Colors<'a>, PixelPoints) {
```

Comment:
I don't think it's necessary to also create `PixelPoints` in the `RleXColors` constructors. Instead `PixelPoints` should have a constructor like: `PixelPoints::new(image_size: Size, row_order: RowOrder)`, which would be usable for all image types in `RawPixels::new`.